src/AssemblerReads.cpp

```diff
@@ -183,14 +186,79 @@ void Assembler::writeReadsSummary()
         csv << double(vertexCount) / double(markerCount) << ",";
 
         // Oxford Nanopore standard metadata.
-        csv << reads.getMetaData(readId, "runid") << ",";
-        csv << reads.getMetaData(readId, "sampleid") << ",";
-        csv << reads.getMetaData(readId, "read") << ",";
-        csv << reads.getMetaData(readId, "ch") << ",";
-        csv << reads.getMetaData(readId, "start_time") << ",";
+        csv << reads->getMetaData(readId, "runid") << ",";
+        csv << reads->getMetaData(readId, "sampleid") << ",";
+        csv << reads->getMetaData(readId, "read") << ",";
+        csv << reads->getMetaData(readId, "ch") << ",";
+        csv << reads->getMetaData(readId, "start_time") << ",";
 
         // End of the line for this read.
         csv << "\n";
-     }
+    }
+}
+
+
+uint64_t Assembler::adjustCoverageAndGetNewMinReadLength(uint64_t desiredCoverage) {
+    cout << timestamp << "Adjusting for desired coverage." << endl;
+    cout << "Desired Coverage: " << desiredCoverage << endl;
+    uint64_t cumulativeBaseCount = reads->getTotalBaseCount();
+
+    if (desiredCoverage > cumulativeBaseCount) {
+        assemblerInfo->minReadLength = 0ULL;
+        return assemblerInfo->minReadLength;
+    }
+
+    assemblerInfo->minReadLength = std::numeric_limits<uint64_t>::max();
+
+    const auto& histogram = reads->getReadLengthHistogram();
+    for (uint64_t length = 0; length < histogram.size(); length++) {
+        const uint64_t frequency = histogram[length];
+        if (frequency) {
+            const uint64_t baseCount = frequency * length;
+            if (cumulativeBaseCount > desiredCoverage) {
+                cumulativeBaseCount -= baseCount;
+                continue;
+            }
+
+            assemblerInfo->minReadLength = length;
```

Comment:
The loop ends when `cumulativeBaseCount` drops below `desiredCoverage` (equality included). So in most cases actual coverage will be a bit below what the user specified. I think it would be better instead to set `minReadLength` to the lowest value such that `cumulativeBaseCount` was greater or equal to `desiredCoverage`. That way actual coverage would be a bit more than the user specified (or the same), rather than a bit less. 

You can make this change in a separate PR.

Reply:
With that change, assuming there is at least one read to begin with, you would no longer have to deal with the case `minReadLength ==std::numeric_limits<uint64_t>::max()` in `main`.